app/src/main/java/io/zenandroid/onlinego/ui/screens/puzzle/TsumegoUI.kt

```diff
@@ -0,0 +1,249 @@
+import io.zenandroid.onlinego.ui.composables.RatingBar
+import io.zenandroid.onlinego.ui.screens.puzzle.TsumegoAction.*
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filter
+import org.commonmark.node.*
+import org.koin.core.context.GlobalContext
+import java.time.temporal.ChronoUnit.*
+
+private const val TAG = "TsumegoUI"
+
+@Composable
+fun TsumegoScreen(
+    state: TsumegoState,
+    hasPreviousPuzzle: Boolean,
+    hasNextPuzzle: Boolean,
+    onMove: (Cell) -> Unit,
+    onHint: () -> Unit,
+    onResetPuzzle: () -> Unit,
+    onRate: (Int) -> Unit,
+    onSolved: () -> Unit,
+    onPreviousPuzzle: () -> Unit,
+    onNextPuzzle: () -> Unit,
+    onBack: () -> Unit,
+) {
+    val solved = state.continueButtonVisible
+    Column (
+        modifier = Modifier.fillMaxHeight()
+    ) {
+        val titleState = remember {
+            val base = "Tsumego"
+            derivedStateOf {
+                state.puzzle?.name?.let {
+                    "${base}: ${it}"
+                } ?: base
+            }
+        }
+        TopAppBar(
+            title = {
+                Text(
+                    text = titleState.value,
+                    fontSize = 18.sp,
+                    modifier = Modifier.weight(1f)
+                )
+                if(state.solutions.size > 0) {
+                    Image(painter = painterResource(R.drawable.ic_check_circle),
+                        modifier = Modifier
+                            .padding(start = 18.dp),
+                        contentDescription = "Solved"
+                    )
+                }
+            },
+            elevation = 1.dp,
+            navigationIcon = {
+                IconButton(onClick = { onBack() }) {
+                    Icon(imageVector = Icons.Default.ArrowBack, contentDescription = null)
+                }
+            },
+            backgroundColor = MaterialTheme.colors.surface
+        )
+
+        state.puzzle?.let {
+            Surface(
+                shape = MaterialTheme.shapes.medium,
+                modifier = Modifier.fillMaxSize()
+            ) {
+              //val listener = { action: TsumegoAction ->
+              //    Toast.makeText(requireContext(), action.toString(), Toast.LENGTH_SHORT).show()
+              //    internalActions.onNext(action)
+              //}
+                Column(modifier = Modifier
+                        .padding(horizontal = 10.dp, vertical = 10.dp)) {
+                    it.puzzle.let {
+                        Board(
+                            boardWidth = it.width,
+                            boardHeight = it.height,
+                            position = state.boardPosition,
+                            boardTheme = state.boardTheme,
+                            drawCoordinates = state.drawCoordinates,
+                            interactive = state.boardInteractive,
+                            drawShadow = false,
```

Comment:
why no shadow? That feature was added to increase performance for thumbnail boards presented in lists.

Reply:
I actually don't remember why I set this. I'll add it back unless I remember why